reporting/src/request/steps/token-telemetry/cached-entry-pipeline.js

```diff
@@ -83,27 +83,30 @@ export default class CachedEntryPipeline {
    * @param batchInterval how often to run batches
    * @param batchLimit maximum messages per batch
    */
-  async init(
-    inputObservable,
-    sendMessage,
-    batchInterval,
-    batchLimit,
-    overflowSubject,
-  ) {
+  async init(inputObservable, sendMessage, batchInterval, batchLimit) {
     await this.cache.isReady;
     const pipeline = new Subject();
+    const retryQueue = new Subject();
     this.input = inputObservable;
 
     let batch = [];
     setInterval(() => {
-      pipeline.pub(batch);
-      batch = [];
+      if (batch.length > 0) {
+        pipeline.pub(batch);
+        batch = [];
+      }
     }, batchInterval);
 
     inputObservable.subscribe((token) => {
       batch.push(token);
     });
 
+    retryQueue.subscribe((token) => {
+      setTimeout(() => {
```

Comment:
I wonder if that is a robust strategy to push with a 1ms delay. Could it not still overflow when the timeout hits and all the push operations are executed?